instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/db/DbSemanticConventions.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.db;
+
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+
+public final class DbSemanticConventions {
+  private static final String DB_QUERY = "DB Query";
+
+  /**
+   * Returns a {@link SpanNameExtractor} that constructs the span name according to DB semantic
+   * conventions: {@code <db.operation> <db.name><table>}.
+   *
+   * @see DbAttributesExtractor#dbOperation(Object) used to extract {@code <db.operation>}.
+   * @see DbAttributesExtractor#dbName(Object) used to extract {@code <db.name>}.
+   * @see SqlAttributesExtractor#dbTable(Object) used to extract {@code <db.table>}.
+   */
+  public static <REQUEST> SpanNameExtractor<REQUEST> spanName(
+      DbAttributesExtractor<REQUEST> attributesExtractor) {
+    return spanName(attributesExtractor, request -> DB_QUERY);
+  }
+
+  /**
+   * A helper method for constructing the span name formatting according to DB semantic conventions:
+   * {@code <db.operation> <db.name><table>}. If {@code dbName} and {@code operation} are not
+   * provided then {@code defaultSpanNameExtractor} is used to compute the span name.
+   *
+   * @see DbAttributesExtractor#dbOperation(Object) used to extract {@code <db.operation>}.
+   * @see DbAttributesExtractor#dbName(Object) used to extract {@code <db.name>}.
+   * @see SqlAttributesExtractor#dbTable(Object) used to extract {@code <db.table>}.
+   */
+  public static <REQUEST> SpanNameExtractor<REQUEST> spanName(
+      DbAttributesExtractor<REQUEST> attributesExtractor,
+      SpanNameExtractor<REQUEST> defaultSpanNameExtractor) {
```

Comment:
I think that Cassandra uses `db.statement` as the default - maybe it can be avoided, but let's revisit this topic once we get to the Cassandra instrumentation. I'll remove this option in the meantime - let's keep this as simple as possible.